dbt/adapters/dremio/api/authentication.py

```diff
@@ -67,3 +72,24 @@ def get_headers(self):
             authorization_token=self.pat
         )
         return self.build_headers(authorization_field)
+
+@dataclass
+class Parameters:
+    base_url: str
+    authentication: DremioAuthentication    
+    is_cloud: boolean = True
+    cloud_project_id: Optional[str] = None
+
+def login(api_parameters: Parameters, timeout=10, verify=True):
```

Comment:
Should login just be under endpoints.py with all the other endpoints?

Reply:
I feel as though login is separate enough from the other endpoints that it belongs in authentication. However, this is just personal preference, I can also see what you mean in having all endpoint functions being in endpoints.py. Would love to hear your input @jlarue26.

Reply:
Moved login into endpoints.py.